dashboard/modules/job/utils.py

```diff
@@ -105,9 +109,6 @@ def file_tail_iterator(path: str) -> Iterator[Optional[List[str]]]:
                 # Add line to current chunk
                 lines.append(curr_line)
                 chunk_char_count += len(curr_line)
```

Comment:
here the `time.sleep(1)` is removed making it a busy loop if all logs are read. What about: 

change the signature to return `AsyncIterator[Optional[List[str]]]` or even `AsyncIterator[List[str]]`, and if it reached EOF, do `asyncio.sleep(1)`.

And consequently change `JobLogStorageClient.tail_logs` to return `AsyncIterator`.

In fact this works nicely so `JobManager.tail_job_logs` no longer needs its own `asyncio.sleep`.